src/Serilog.Sinks.Exceptionless/Sinks/Exceptionless/ExceptionlessSink.cs

```diff
@@ -88,8 +90,44 @@ public class ExceptionlessSink : ILogEventSink, IDisposable {
 
             if (_includeProperties && logEvent.Properties != null) {
                 foreach (var property in logEvent.Properties)
-                    if (property.Key != Constants.SourceContextPropertyName)
-                        builder.SetProperty(property.Key, property.Value.FlattenProperties());
+                {
+                    if (property.Key == Constants.SourceContextPropertyName)
+                    {
+                        continue;
+                    }
+                    else if (property.Key == Event.KnownDataKeys.UserInfo)
+                    {
+                        if (property.Value is StructureValue structure && structure.TypeTag == nameof(UserInfo))
```

Comment:
Since the value is already of the correct type isn't there a way to cast to it and then just read the properties right form the value instead of looking at the structure properties and doing a ton of extra work?

Reply:
I'm still concerned by the extra work going on here. I don't understand why the object needs to be recreated or casting a sequence to object value. Can you please let me know why we need to do this?

Reply:
StructureValue 
![image](https://user-images.githubusercontent.com/2100585/100823757-c5e4b580-348f-11eb-879e-619d71736ec6.png)

Reply:
I wish we could have a discussion on this with some explanation so I can understand why the code is the way it is but I guess I just need to pull it down and run it. I don't understand why we need to do the extra work to reconstruct the object we are setting on the log event.

Reply:
![image](https://user-images.githubusercontent.com/2100585/100827261-4a870200-3497-11eb-9a80-c38eb65a2315.png)
Sorry, I did not implement it to tell you why I did this. First of all, I saw the UserSession function in Exceptionless, but we used serilog-sinks-exceptionless to write data, and found that user information cannot be written in serilog Into the exceptionless UserSession, so this PR
Debug to find serilog-sinks-exceptionless as long as you intercept Properties in the Emit method and then pass the intercepted data through EventBuilder.SetUserDescription, you can pass the specified data into the exceptionless UserSession
At the beginning, I wanted to force StructureValue into an object. I found that reflection was needed, which consumes more resources. I modified it to nameof and initialized the object. The use of nameof is to consider whether this object will be modified later. I think I am overly worried.
I just changed it again, removed the nameof and the object, please take the time to take a look!
Thank you!
By the way, I have pushed this nuget to the private Nuget, and it is currently being used in a production environment!

Reply:
Awesome, thanks for the update. I think the name of is definitely better for readability. I'll take a look at this soon :-).